jekyll/_posts/2021-01-19-community-survey-results-2020.md

```diff
@@ -0,0 +1,221 @@
+Three quarters of people who are allowed to use Nim at work plan to use it in 2021.
+
+People mention stability, better tooling, more 3rd party libraries, having to mature, and better documentation as the main fields where Nim has to improve to be more accepted in their companies.
+
+
+
+
+## Learning Nim
+
+<img src="{{site.baseurl}}/assets/news/images/survey2020/17.png" alt="What learning resources, if any, did you use to learn Nim?" style="width:100%"/>
+
+Almost all of our users have read [the official tutorials](https://nim-lang.org/docs/tut1.html), about half of them have read ["Nim by Example"](https://nim-by-example.github.io/getting_started/) and/or ["Nim in Action"](https://book.picheta.me) book.
+[Rosetta code examples](http://rosettacode.org/wiki/Category:Nim) are also a popular choice as a learning resource, followed by ["Nim Basics"](https://narimiran.github.io/nim-basics/), ["Nim Days"](https://xmonader.github.io/nimdays/) and ["Nim Notes"](https://scripter.co/notes/nim).
+
+
+<img src="{{site.baseurl}}/assets/news/images/survey2020/18.png" alt="What kind of learning materials does Nim need?" style="width:100%"/>
+
+If anyone would like to create some more learning materials for Nim (which is very high on the community priority list, see below under the "Nim in 2021 and beyond" section), the most wanted types of learning materials are code examples and written tutorials.
+
+
+
+## Nim community and contributions
+
+<img src="{{site.baseurl}}/assets/news/images/survey2020/19.png" alt="Have you made code contributions to the Nim project in the past?" style="width:100%"/>
+
+Compared to 2019, we see a 20% increase of people who have already contributed to Nim!
+And there are almost three times more people who haven't contributed to Nim so far but plan to do that in the future.
+We're looking forward to this :)
+
+The most frequent reasons for not contributing (yet) are: lack of time and lack of skill/experience.
+
+
+<img src="{{site.baseurl}}/assets/news/images/survey2020/19a.png" alt="Did you make any donations to Nim?" style="width:100%"/>
+
+In the past we haven't stressed enough the importance of the [community donations](https://nim-lang.org/donate.html), which is reflected in this graph: only 10% of our users have made a donation in the past.
+We are happy to see that 2/3 of our users plan to do it in the future, which will help us to have more manpower to fix the most important bugs more promptly and work on some exciting new features.
+
+
+
+
+
+## Nim tools
+
+We've asked you how satisfied you are with the Nim tooling:
+
+<img src="{{site.baseurl}}/assets/news/images/survey2020/29.png" alt="What is your experience with the following tools? [nimble]" style="width:100%"/>
+<img src="{{site.baseurl}}/assets/news/images/survey2020/30.png" alt="What is your experience with the following tools? [choosenim]" style="width:100%"/>
+
+Most of our users use Nimble package manager and they are very satisfied by it.
```

Comment:
I'm curious how many who use nim personally are `very satisfied` because I don't see how you can use Nimble for professional projects that require version pinning at the moment so that builds are somewhat reproducible.

Reply:
What do you consider "use nim personally"? The survey asks people what their thoughts are on Nimble and there is an option to state "I haven't used it". It's unlikely someone would use Nimble if they aren't personally using Nim.

I know this might go against your personal experience, but most Nim users aren't yet writing big enough applications for lock files to matter. In addition, Nimble does support version pinning already, I use it all the time for my applications, it's not as good as having Nimble generate the pinned versions for you but it works well enough.

It's annoying to see the loud voices saying that Nimble sucks when the survey has pretty consistently shown that practically all of the community is satisfied with it. We need to invest development time into it to keep it that way though, it's been a while since its seen significant development put into it.